com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/IAdditionalExportService.java

```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.resource;
+
+import com.google.inject.ImplementedBy;
+
+
+/**
+ * This service allows additional export rules for a given DSL to be defined outside of its plug-in.
+ */
+@ImplementedBy(DefaultAdditionalExportService.class)
+public interface IAdditionalExportService extends IAdditionalExport {

```

Comment:
it looks strange to me that DefaultAdditionalExportService implements IAdditionalExport. I'd expect that the contributions should implement IAdditionalExport and the services would just look them up and inject members. However, it is already implemented this way with additional inferrers, so we can live with this.

Reply:
Well. it's that the contributions and the service that aggregates them for the extension point in a DSL feature have the same API. I was not sure about this patter either. But duplicating the definition seems to be even worse.